enterprise/backend/src/metabase_enterprise/workspaces/api.clj

```diff
@@ -288,6 +290,30 @@
       (t2/hydrate :contents)
       ws->response))
 
+(api.macros/defendpoint :get "/:id/log"
+  :- [:map
+      [:workspace_id ms/PositiveInt]
+      [:status [:enum :creating :ready]]
+      [:logs [:sequential [:map
+                           [:id ms/PositiveInt]
+                           [:workspace_id ms/PositiveInt]
+                           [:task :keyword]
+                           [:started_at :any]
+                           [:completed_at [:maybe :any]]
+                           [:result [:maybe :keyword]]
+                           [:message [:maybe :string]]]]]]
+  "Get workspace creation status and recent log entries for polling during async setup"
+  [{:keys [id]} :- [:map [:id ms/PositiveInt]]
+   _query-params]
+  (let [workspace (api/check-404 (t2/select-one :model/Workspace :id id))
+        logs      (t2/select :model/WorkspaceLog
```

Comment:
nit: I prefer for all APIs to be explicit about what database details they expose